app/org/sagebionetworks/bridge/services/AuthenticationServiceImpl.java

```diff
@@ -353,11 +191,12 @@ public UserSession getSessionFromAccount(Study study, Account account) {
         user.setStudyKey(study.getIdentifier());
 
         final String healthCode = getHealthCode(study, account);
-        user.setHealthCode(healthCode);
-
-        user.setSignedMostRecentConsent(consentService.hasUserSignedMostRecentConsent(user, study));
-        user.setConsent(consentService.hasUserConsentedToResearch(user, study));
-        user.setDataSharing(optionsService.getBooleanOption(healthCode, Option.DATA_SHARING));
+        if (healthCode != null) {
+            user.setHealthCode(healthCode);
+            user.setDataSharing(optionsService.getBooleanOption(healthCode, Option.DATA_SHARING));
+        }
```

Comment:
You're right it should be unnecessary. It was there to start with a different form, but I'll remove it and rerun tests.